### DISCUSSION TITLE
How can I access real-time data for advertising reports?

### DISCUSSION THREAD

**Discussion by xiaozhong1995** [amzn/ads-advanced-tools-docs#236] [General]
Hello！

The current reports are created through APIs and then downloaded, with a significant time delay. 

I would like to know if Amazon provides a way to access report data in real-time by setting up a subscription to have Amazon push data to my app in real-time.

**Comment by kdwivedi13** [comment]
Amazon Marketing Stream delivers updates hourly, giving you a constant pulse on your performance.  For details, check out the official [documentation](https://advertising.amazon.com/API/docs/en-us/guides/amazon-marketing-stream/overview).  To streamline your integration, explore the [Amazon Marketing Stream reference application](https://github.com/amzn/amazon-marketing-stream-examples) as well. There's even a helpful [demo on YouTube](https://www.youtube.com/watch?v=_4o6QuTZxxs&t=120s) to see it in action.

**Reply by xiaozhong1995** [reply]
Thank you, your answer is very helpful to me.

However, I have another question about Amazon Marketing Stream. I have reviewed the corresponding documentation, and it supports fewer datasets compared to the Reporting API. Will it add more sales metrics in the future?

**Comment by kdwivedi13** [comment]
I'm pleased to assist you further. Before we delve into your current query, I'd like to revisit your previous question to gain a deeper understanding of your concerns. Firstly, could you elaborate on your use case and explain why you perceive the report downloading process to be slow? Additionally, it would be beneficial if you could create a [JIRA ticket](https://amzn-clicks.atlassian.net/servicedesk/customer/portal/2/group/2) detailing the client ID and the report API request payload that is experiencing extended generation times. We are keen to investigate this matter more thoroughly.

Regarding your recent question, I want to assure you that we are continuously expanding the datasets available in the Amazon Marketing Stream. However, if there's a specific report you feel is currently missing and could enhance your insights, please do let us know. Your feedback is invaluable in helping us identify and prioritize additions that would be most beneficial to our users. For discussions on new metrics or any further suggestions, feel free to initiate a conversation using this [link](https://github.com/amzn/ads-advanced-tools-docs/discussions/new?category=ideas).
